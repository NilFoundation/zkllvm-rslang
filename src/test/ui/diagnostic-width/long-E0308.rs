// compile-flags: --diagnostic-width=60
// normalize-stderr-test: "long-type-\d+" -> "long-type-hash"

struct Atype<T, K>(T, K);
struct Btype<T, K>(T, K);
struct Ctype<T, K>(T, K);

fn main() {
    let x: Atype<
      Btype<
        Ctype<
          Atype<
            Btype<
              Ctype<
                Atype<
                  Btype<
                    Ctype<i32, i32>,
                    i32
                  >,
                  i32
                >,
                i32
              >,
              i32
            >,
            i32
          >,
          i32
        >,
        i32
      >,
      i32
    > = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
        Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
            Ok("")
        ))))))))))))))))))))))))))))))
    ))))))))))))))))))))))))))))));
    //~^^^^^ ERROR E0308

    let _ = Some(Ok(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(
        Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(
            Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(Some(
                Some(Some(Some(Some(Some(Some(Some(Some(Some("")))))))))
            )))))))))))))))))
        ))))))))))))))))))
    ))))))))))))))))) == Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
        Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
            Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
        ))))))))))))))))))))))))))))))
    ))))))))))))))))))))))));
    //~^^^^^ ERROR E0308

    let x: Atype<
      Btype<
        Ctype<
          Atype<
            Btype<
              Ctype<
                Atype<
                  Btype<
                    Ctype<i32, i32>,
                    i32
                  >,
                  i32
                >,
                i32
              >,
              i32
            >,
            i32
          >,
          i32
        >,
        i32
      >,
      i32
    > = ();
    //~^ ERROR E0308

    let _: () = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
        Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(
            Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
        ))))))))))))))))))))))))))))))
    ))))))))))))))))))))))));
    //~^^^^^ ERROR E0308
}
