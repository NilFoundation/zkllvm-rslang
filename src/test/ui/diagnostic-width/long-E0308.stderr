error[E0308]: mismatched types
  --> $DIR/long-E0308.rs:33:9
   |
LL |        let x: Atype<
   |  _____________-
LL | |        Btype<
LL | |          Ctype<
LL | |            Atype<
...  |
LL | |        i32
LL | |      > = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok...
   | | _____-___^
   | ||_____|
   |  |     expected due to this
LL |  |         Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok...
LL |  |             Ok("")
LL |  |         ))))))))))))))))))))))))))))))
LL |  |     ))))))))))))))))))))))))))))));
   |  |__________________________________^ expected struct `Atype`, found enum `Result`
   |
   = note: expected struct `Atype<Btype<..., ...>, ...>`
           the full type name has been written to '$TEST_BUILD_DIR/diagnostic-width/long-E0308/long-E0308.long-type-hash.txt'
                found enum `Result<Result<..., ...>, ...>`
           the full type name has been written to '$TEST_BUILD_DIR/diagnostic-width/long-E0308/long-E0308.long-type-hash.txt'

error[E0308]: mismatched types
  --> $DIR/long-E0308.rs:46:26
   |
LL |       ))))))))))))))))) == Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(O...
   |  __________________________^
LL | |         Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(...
LL | |             Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
LL | |         ))))))))))))))))))))))))))))))
LL | |     ))))))))))))))))))))))));
   | |____________________________^ expected enum `Option`, found enum `Result`
   |
   = note: expected enum `Option<Result<..., ...>>`
           the full type name has been written to '$TEST_BUILD_DIR/diagnostic-width/long-E0308/long-E0308.long-type-hash.txt'
              found enum `Result<Result<..., ...>, ...>`
           the full type name has been written to '$TEST_BUILD_DIR/diagnostic-width/long-E0308/long-E0308.long-type-hash.txt'

error[E0308]: mismatched types
  --> $DIR/long-E0308.rs:77:9
   |
LL |       let x: Atype<
   |  ____________-
LL | |       Btype<
LL | |         Ctype<
LL | |           Atype<
...  |
LL | |       i32
LL | |     > = ();
   | |     -   ^^ expected struct `Atype`, found `()`
   | |_____|
   |       expected due to this
   |
   = note: expected struct `Atype<Btype<..., ...>, ...>`
           the full type name has been written to '$TEST_BUILD_DIR/diagnostic-width/long-E0308/long-E0308.long-type-hash.txt'
           found unit type `()`

error[E0308]: mismatched types
  --> $DIR/long-E0308.rs:80:17
   |
LL |       let _: () = Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(O...
   |  ____________--___^
   | |            |
   | |            expected due to this
LL | |         Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(Ok(...
LL | |             Ok(Ok(Ok(Ok(Ok(Ok(Ok("")))))))
LL | |         ))))))))))))))))))))))))))))))
LL | |     ))))))))))))))))))))))));
   | |____________________________^ expected `()`, found enum `Result`
   |
   = note: expected unit type `()`
                   found enum `Result<Result<..., ...>, ...>`
           the full type name has been written to '$TEST_BUILD_DIR/diagnostic-width/long-E0308/long-E0308.long-type-hash.txt'

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0308`.
